src/label/labelLayoutHelper.ts

```diff
@@ -315,6 +315,13 @@ export function hideOverlap(labelList: LabelLayoutInfo[]) {
         const transform = labelItem.transform;
         const label = labelItem.label;
         const labelLine = labelItem.labelLine;
+
+        // if the label has no text, remove it as a candidate for overlap checking
+        if (label.style?.text === '') {
```

Comment:
After discussing with other members, we tend to believe it's better to make the change here instead of in ZRender because it may be controversial to define whether it's overlapping for an object with empty height/width inside other elements. So we still need to do the change here. I'll close the ZRender PR.

I think the overall logic is great, only that it may make greater sense if we could check the width and height of the text here instead of checking `text === ''`. And we try to use `?.` in ECharts to reduce the compiled size, so `label.style && lable.style.text` is a better way to check like this.

Thanks for your contribution!

Reply:
Thank you for the update.

I updated the conditional to check the width and height of the bounding rectangle instead of checking the label's text itself.

Let me know if the updated conditional is what you were expecting.

Thank you for your help, Ovilia!